idaes/generic_models/properties/core/pure/Perrys.py

```diff
@@ -146,32 +146,59 @@ class dens_mol_liq_comp():
 
     @staticmethod
     def build_parameters(cobj):
-        cobj.dens_mol_liq_comp_coeff_1 = Var(
-                doc="Parameter 1 for liquid phase molar density",
-                units=pyunits.kmol*pyunits.m**-3)
-        set_param_from_config(cobj, param="dens_mol_liq_comp_coeff", index="1")
-
-        cobj.dens_mol_liq_comp_coeff_2 = Var(
-                doc="Parameter 2 for liquid phase molar density",
-                units=pyunits.dimensionless)
-        set_param_from_config(cobj, param="dens_mol_liq_comp_coeff", index="2")
-
-        cobj.dens_mol_liq_comp_coeff_3 = Var(
-                doc="Parameter 3 for liquid phase molar density",
-                units=pyunits.K)
-        set_param_from_config(cobj, param="dens_mol_liq_comp_coeff", index="3")
-
-        cobj.dens_mol_liq_comp_coeff_4 = Var(
-                doc="Parameter 4 for liquid phase molar density",
-                units=pyunits.dimensionless)
-        set_param_from_config(cobj, param="dens_mol_liq_comp_coeff", index="4")
+        if cobj._name in ("water", "sulfuric_acid", "o-terphenyl"):
```

Comment:
I don't think using specific component names is the correct way to do this. First of all, it is too fragile (e.g. you used "water" where we generally use "H2O" so this would break).

I think we need a flag of some form as an argument to the method which users set in the config dict. E.g.:

```
try:
   flag = cobj.config.parameter_data["dens_mol_liq_comp_form"]
except KeyError:
   flag = DEFAULT_VALUE
```